Icu 6794 refactor attributes using the newer is nested attribute annotation

ZedLi opened the pull request:
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-6794)

## Description

:technologist: [Admin preview](https://boundary-ui-git-icu-6794-refactor-attributes-u-0c5223-hashicorp.vercel.app/scopes/s_id6ozx3wue/auth-methods/am_wtbbeizy0q)

This PR aims to remove the usage of attribute fragments in auth methods and hoist them up to the root of the model. This needs to be done before we remove the usage of fragments with the arrays.

There will be a followup PR to cleanup the serializer for normalizing errors.

calcaide commented on addons/api/addon/generated/models/auth-method.js:
```diff
@@ -49,4 +49,26 @@ export default class GeneratedAuthMethodModel extends BaseModel {
+  @attr('string', { isNestedAttribute: true }) client_secret;
+
+  @attr('string', { readOnly: true, isNestedAttribute: true })
+  client_secret_hmac;
```
Nevermind, that field was part of the fragment, now I see it on the fragment model 😉

calcaide reviewed (approved):
LGTM! Amazing work @ZedLi

randallmorey reviewed (approved):
Thanks for the conversion.  I found only attributes tests validating the `state` field.  Should we take this opportunity to add tests for all of the nested attribute fields?

ZedLi commented:
> Thanks for the conversion. I found only attributes tests validating the `state` field. Should we take this opportunity to add tests for all of the nested attribute fields?

Added more acceptance tests for creating and updating the OIDC form. Also updated the normalizer test to include the fields
